stack-graphs/src/cycles.rs

```diff
@@ -127,3 +135,150 @@ where
         true
     }
 }
+
+// ----------------------------------------------------------------------------
+// Cycle detector
+
+pub trait Appendable {
+    type Ctx;
+
+    fn append_to(
+        &self,
+        graph: &StackGraph,
+        partials: &mut PartialPaths,
+        ctx: &mut Self::Ctx,
+        path: &mut PartialPath,
+    ) -> Result<(), PathResolutionError>;
+    fn start_node(&self, ctx: &mut Self::Ctx) -> Handle<Node>;
+    fn end_node(&self, ctx: &mut Self::Ctx) -> Handle<Node>;
+}
+
+#[derive(Clone)]
+pub struct AppendingCycleDetector<A> {
+    appendages: List<A>,
```

Comment:
One thing that is confusing me a bit is how you're using `appendages` both as "a list of `A`" and also as "the arena where the list's cells are stored".  Could this field instead be `appendables`?  That would align with the `A: Appendable` constraint below, and would leave `appendages` for the arena.

Reply:
Thanks for that suggestion! I was struggling a bit with the naming there.